experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,275 @@
+    Y,_ = germ_at_point(localized_ring(OO(X)))
+    return Y
+end
+
+@attr SpaceGerm function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return X
+end
+
+############################################################################################################
+# allow user to specify point also as ideal
+############################################################################################################
+
+function _maxideal_to_point(I::MPolyIdeal)
+  R=base_ring(I)
+  o = degrevlex(gens(R))
+  G=groebner_basis(I)
+  LG = leading_ideal(I;ordering=o)
+  dim(LG)==0 || error("Ideal does not describe finite set of points")
+  vd,vbasis = _vdim_hack(LG)
+  vd ==1 || error("Ideal does not describe a single K-point")
+  return [leading_coefficient(normal_form(v,G)) for v in gens(R)]
+end
+
+### _vdim_hack only to be used until vdim of 0-dimensional ideals is implemented properly
+### _vdim_hack assumes for simplicity that 
+###         1. dim(I)=0 has been tested, 
+###         2.I is a monomial ideal
+function _vdim_hack(I::MPolyIdeal)
+  leer=elem_type(base_ring(I))[]
+  R = base_ring(I)
+  if one(R) in I 
+    return 0,leer
+  end
+  G=groebner_basis(I)
+  M = ideal(R,gens(R))
+  result=[R(1)]
+  J = ideal(R,normal_form(gens(M),I))
+  while dim(J) != length(gens(R))
+    Jtemp = leer
+    JN=gens(J)
+    for i in 1:length(JN)
+      if(JN[i]!=R(0))
+        push!(result,JN[i])
+        push!(Jtemp, JN[i])
+      end
+    end
+    J = ideal(R,Jtemp)
+    J = ideal(R,normal_form(gens(J*M),I))
+  end
+  return length(result),result
+end
+
+############################################################################################################
+### constructors
+############################################################################################################
+function SpaceGerm(X::AbsSpec, a::Vector)
+  R = ambient_ring(X)
+  kk = coefficient_ring(R)
+  b = [kk.(v) for v in a]  ## throws an error, if vector entries are not compatible
+  U = MPolyComplementOfKPointIdeal(R,b)
+  Y = Spec(Localization(OO(X), U)[1])
+  Z = SpaceGerm(Y)
+  set_attribute!(Z,:representative,X)
+  return SpaceGerm(Y)
+end
+
+function SpaceGerm(X::AbsSpec, I::MPolyIdeal)
+  R = base_ring(I)
+  R === ambient_ring(X) || error("rings are not compatible")
+  a = _maxideal_to_point(I)
+  Y = SpaceGerm(X,a)
+  return Y
+end
+
+to_poly_ideal(I::MPolyQuoIdeal) = ideal(base_ring(base_ring(I)),lift.(gens(I))) + modulus(base_ring(I))
+to_poly_ideal(I::MPolyLocalizedIdeal) = ideal(base_ring(base_ring(I)), gens(saturated_ideal(I)))
+to_poly_ideal(I::MPolyQuoLocalizedIdeal) = ideal(base_ring(base_ring(I)),gens(saturated_ideal(I))) + modulus(quotient_ring(base_ring(I)))
+
+function SpaceGerm(X::AbsSpec, I::Ideal)
+  A = base_ring(I)
+  A === OO(X) || error("rings are incompatible")
+  J = to_poly_ideal(I)
+  a = _maxideal_to_point(J)
+  Y = SpaceGerm(X,a)
+  return Y
+end
```

Comment:
This should work. However, the `saturated_ideal` has to compute a primary decomposition (for localizations at points), while we could easily do the checks using local orderings. This might not be of primary interest here, since we do not expect `X` to already be a germ, but either way, I think it's more sustainable, to directly hook this up with the respective local(ized) rings and their ideals. Since this is very difficult for somebody who hasn't programmed the localizations themselves, I suggest we sit down together on Thursday to take care of this.

Reply:
Oh, now I see that saturated_ideal does not implement the shortcut "1 in I && return ideal(R,1)" (or the respective radical membership test) before passing to the primary decomposition. You are right, we should discuss how and where to do this stuff properly tomorrow.